content_handler/vulkan_surface_pool.h

```diff
@@ -0,0 +1,77 @@
+// Copyright 2017 The Fuchsia Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#pragma once
+
+#include <list>
+#include <unordered_map>
+#include "flutter/content_handler/vulkan_surface.h"
+#include "lib/ftl/macros.h"
+
+namespace flutter_runner {
+
+class VulkanSurfacePool {
+ public:
+  static const size_t kMaxSurfacesOfSameSize = 3;
+  static const size_t kMaxSurfaceAge = 3;
+
+  VulkanSurfacePool(vulkan::VulkanProcTable& vk,
+                    sk_sp<GrContext> context,
+                    sk_sp<GrVkBackendContext> backend_context,
+                    mozart::client::Session* mozart_session);
+
+  ~VulkanSurfacePool();
+
+  std::unique_ptr<flow::SceneUpdateContext::SurfaceProducerSurface>
+  AcquireSurface(const SkISize& size);
+
+  void SubmitSurface(
+      std::unique_ptr<flow::SceneUpdateContext::SurfaceProducerSurface>
+          surface);
+
+  void AgeAndCollectOldBuffers();
+
+ private:
+  using SurfacesSet = std::list<
+      std::unique_ptr<flow::SceneUpdateContext::SurfaceProducerSurface>>;
+
+  template <class T>
+  static void HashCombine(size_t& seed, T const& v) {
```

Comment:
I agree. But didnt know which one to add it to. The hash probably doesn't need to be this fancy too.